pymeasure/instruments/lakeshore/lakeshore211.py

```diff
@@ -0,0 +1,205 @@
+        | 'celsius'   | Celsius      |
+        +-------------+--------------+
+        | 'sensor'    | Sensor Units |
+        +-------------+--------------+
+        | 'fahrenheit'| Fahrenheit   |
+        +-------------+--------------+
+        """,
+        values={'kelvin': 0, 'celsius': 1, 'sensor': 2, 'fahrenheit': 3},
+        map_values=True
+    )
+
+    temperature_celsius = Instrument.measurement(
+        "CRDG?",
+        """Reads the temperature of the sensor in celsius
+        """
+    )
+
+    temperature_fahrenheit = Instrument.measurement(
+        "FRDG?",
+        """Reads the temperature of the sensor in fahrenheit
+        """
+    )
+
+    temperature_sensor = Instrument.measurement(
+        "SRDG?",
+        """Reads the temperature of the sensor in sensor units
+        """
+    )
+
+    temperature_kelvin = Instrument.measurement(
+        "KRDG?",
+        """Reads the temperature of the sensor in kelvin
+        """
+    )
+
+    relay = Instrument.control(
+        "RELAY?", "RELAY %d,%d",
+        """
+        Specifies which relay to configure. Values need to be supplied as a tuple of
+        (relay number, relay mode)
+        Relay number can be 1 or 2
+
+        +--------+-----------------+
+        | setting|       mode      |
+        +--------+-----------------+
+        | 0      | low alarm relay |
+        +--------+-----------------+
+        | 1      | high alarm relay|
+        +--------+-----------------+
+
+        Relay mode can be 0, 1, or 2
+
+        +--------+--------+
+        | setting| mode   |
+        +--------+--------+
+        | 0      | off    |
+        +--------+--------+
+        | 1      | on     |
+        +--------+--------+
+        | 2      | alarms |
+        +--------+--------+
+
+        Property is UNTESTED
+        """,
+        cast=int
+    )
+
+    def alarm_status(self):
+        """
+        Query the current alarm status
+
+        :return: List of current status [on, high_value, low_value, deadband, latch]
+        """
+
+        status = self.values('ALARM?')
+        return dict(zip(self.alarm_keys,
+                        [int(status[0]), float(status[1]), float(status[2]), float(status[3]),
+                         int(status[4])]))
+
+    def alarm_config(self, on=True, high_value=270.0, low_value=0.0, deadband=0, latch=False):
+        """Configures the alarm parameters for the input.
+
+        :param on:  Boolean setting of alarm, default True
+        :param high_value: High value the temperature is checked against to activate the alarm
+        :param low_value: Low value the temperature is checked against to activate the alarm
+        :param deadband: Value that the temperature must change outside of an alarm condition
+        :param latch: Specifies if the alarm should latch or not
+        """
+
+        command_string = "ALARM %d,%g,%g,%g,%d" % (on, high_value, low_value, deadband, latch)
+        self.write(command_string)
```

Comment:
Do you intend these to be properties, then the names would be right. For methods, the names should be verbs, indicating the "action". For example: `configure_alarm` for the second method.
The first method could be a property. You do not have to use the property creator `measurement`, you may add the `@property` decorator to make it a property.

Reply:
No I don't intent these to be properties, even with `@property` and `@property.setter` as an option. The thinking is alarm functionality is different from the measurement functionality and because of the number of argument it is best served as a function.

The relay functionality was also moved to a function from `Instrument.control`, when working with a researcher the relay coding was incorrect and requires an input when querying status. Because of the rationale for having alarm functionality is different from measurement functionality, the relay logic was moved to a function.

Added tests for relays and alarms

Reply:
I agree with your reasoning for why to use a method, sounds good. 
We try to use a consistent (as far as possible) naming scheme (verbs for methods), so for the user it's easier to tell at a glance what might be a function and what a property. Hence @bmoneke's advice to call this e.g. `configure_alarm`. You already see this mirrored in your docstring, so it seems natural already :-).